.github/workflows/test-build.yml

```diff
@@ -7,26 +7,94 @@ on:
     branches: [ master, devops ]
 
 jobs:
-  build:
+  mac-build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        dotnet-version: [3.1.301]
+        os:  [macOS-latest]
+        tests-paths: [./BruteShark/PcapAnalyzerTest, ./BruteShark/PcapProcessorTest]
+        projects-paths: [./BruteShark/BruteSharkCli]
+
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Setup .NET Core
```

Comment:
The code of "Setup", "Install Dependencies", "Test" and "Build" are the same, it worth to search a way to use functions for less code duplication and more readability

Reply:
they are indeed similar, bit there are differences between them, the only two steps that are common to all jobs are the checkout and the setup dotnet,
there is an option to put these two steps in another yaml and create a "composite" actions flow
https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/